File: Булевы функции/js/task1.js
function generateBooleanFunction(n) {
    const numCombinations = 2 ** n;
    return Array.from({ length: numCombinations }, () => Math.floor(Math.random() * 2));
}

function generateTable() {
    const n = parseInt(document.getElementById('nInput').value);
    if (isNaN(n) || n < 1) {
        alert("Введите корректное значение n (≥1)");
        return;
    }

    const booleanFunction = generateBooleanFunction(n);
    const output = document.getElementById('output');
    const tableContainer = document.getElementById('tableContainer');

    // Очистка предыдущих результатов
    output.innerHTML = '';
    tableContainer.innerHTML = '';

    // Вывод строки f
    output.innerHTML = `f = (${booleanFunction.join(', ')})`;

    // Создание таблицы
    const table = document.createElement('table');
    const headerRow = table.insertRow();

    // Заголовки столбцов
    for (let i = 1; i <= n; i++) {
        const th = document.createElement('th');
        th.textContent = `x${i}`;
        headerRow.appendChild(th);
    }
    const thResult = document.createElement('th');
    thResult.textContent = 'f';
    headerRow.appendChild(thResult);

    // Заполнение данных
    for (let i = 0; i < booleanFunction.length; i++) {
        const tr = table.insertRow();
        const binary = i.toString(2).padStart(n, '0');

        // Входные значения
        for (let bit of binary) {
            const td = tr.insertCell();
            td.textContent = bit;
        }

        // Результат
        const tdResult = tr.insertCell();
        tdResult.textContent = booleanFunction[i];
    }

    tableContainer.appendChild(table);
}

File: Булевы функции/python/task1.py
import random

def generate_boolean_function(n):
    # Количество возможных комбинаций входных значений
    num_combinations = 2 ** n

    # Создаем список для хранения значений булевой функции
    boolean_function = [random.randint(0, 1) for _ in range(num_combinations)]

    return boolean_function

def print_truth_table(n, boolean_function):
    num_combinations = 2 ** n # Количество возможных комбинаций входных значений
    print("Таблица истинности для булевой функции:")
    for i in range(num_combinations):
        binary = bin(i)[2:].zfill(n) # Преобразуем индекс в двоичное представление
        args = [int(bit) for bit in binary] # Разделяем двоичное представление на отдельные аргументы
        result = boolean_function[i] # Получаем значение функции

        print(f"Вход: {args} -> Выход: {result}")

    # Вывод в f строку     
    result_str = ", ".join(map(str, boolean_function))
    print(f"f = ({result_str})")
    

n = int(input("Введите n: ")) # Количество аргументов
boolean_function = generate_boolean_function(n)
print_truth_table(n, boolean_function)